libs/SalesforceSDK/src/com/salesforce/androidsdk/push/PushMessaging.java

```diff
@@ -105,31 +101,32 @@ public static void register(Context context, UserAccount account) {
     }
 
     /**
-     * Performs GCM and SFDC un-registration from push notifications for the
-     * specified user account.
+     * Performs SFDC un-registration from push notifications for the specified user account.
+     * Performs GCM un-registration only if this is the last account being logged out.
      *
-     * @param context
-     *         Context.
-     * @param account
-     *         User account.
+     * @param context Context.
+     * @param account User account.
+     * @param isLastAccount True - if this is the last logged in account, False - otherwise.
      */
-    public static void unregister(Context context, UserAccount account) {
+    public static void unregister(Context context, UserAccount account, boolean isLastAccount) {
         if (isRegistered(context, account)) {
             setInProgress(context, true, account);
-            final InstanceID instanceID = InstanceID.getInstance(context);
 
-            threadPool.execute(new Runnable() {
-                @Override
-                public void run() {
-                    try {
-                        // avoid running on main thread
-                        instanceID.deleteInstanceID();
-                    } catch (IOException e) {
-                        SalesforceSDKLogger.e(TAG, "Error deleting InstanceID", e);
+            // Deletes InstanceID only if there are no other logged in accounts.
+            if (isLastAccount) {
```

Comment:
Removing `InstanceID` only if this is the last account. This ensures that tokens for the other logged in users remain valid.